java/ingest/src/main/java/sleeper/ingest/IngestRecordsUsingPropertiesSpecifiedMethod.java

```diff
@@ -158,117 +129,97 @@ public static List<FileInfo> ingestFromRecordIterator(
-     * @param localWorkingDirectory    A local directory for temporary files
-     * @param bufferAllocator          A buffer allocator to use during Arrow-based ingest. It may be null, but if it is
-     *                                 needed, a {@link NullPointerException} will be thrown
-     * @param s3AsyncClient            A client to use during asynchronous ingest. It may be null, but if it is needed,
-     *                                 a {@link NullPointerException} will be thrown
-     * @param hadoopConfiguration      An Hadoop configuration to use when writing Parquet files. It may be null, and if
-     *                                 it is, a default configuration will be used
-     * @param sleeperIteratorClassName The name of the Sleeper iterator to apply
-     * @param sleeperIteratorConfig    The configuration of the iterator
-     * @return The relevant ingest coordinator
+     * @param ingestProperties   The ingest properties to use to configure the ingest
+     * @param instanceProperties The instance properties to use to configure the ingest
+     * @param bufferAllocator    A buffer allocator to use during Arrow-based ingest. It may be null, but if it is
+     *                           needed, a {@link NullPointerException} will be thrown
+     * @param s3AsyncClient      A client to use during asynchronous ingest. It may be null, but if it is needed,
+     *                           a {@link NullPointerException} will be thrown
+     * @param recordIterator     The iterator of records to ingest
+     * @return The relevant IngestRecordsFromIterator object
      */
-    public static IngestCoordinator<Record> createIngestCoordinatorFromProperties(
-            ObjectFactory objectFactory,
-            StateStore sleeperStateStore,
+    public static IngestRecordsFromIterator createIngestRecordsFromIteratorWithProperties(
+            IngestProperties ingestProperties,
             InstanceProperties instanceProperties,
-            TableProperties tableProperties,
-            String localWorkingDirectory,
             BufferAllocator bufferAllocator,
             S3AsyncClient s3AsyncClient,
-            Configuration hadoopConfiguration,
-            String sleeperIteratorClassName,
-            String sleeperIteratorConfig) {
-        Schema sleeperSchema = tableProperties.getSchema();
-        Supplier<RecordBatch<Record>> recordBatchFactoryFn;
-        Function<Partition, PartitionFileWriter> partitionFileFactoryFn;
+            CloseableIterator<Record> recordIterator) {
+        IngestCoordinator<Record> ingestCoordinator;
         // Define a factory function for record batches
-        switch (instanceProperties.get(INGEST_RECORD_BATCH_TYPE).toLowerCase(Locale.ROOT)) {
+        String recordBatchType = instanceProperties.get(INGEST_RECORD_BATCH_TYPE).toLowerCase(Locale.ROOT);
+        String fileWriterType = instanceProperties.get(INGEST_PARTITION_FILE_WRITER_TYPE).toLowerCase(Locale.ROOT);
```

Comment:
I feel like we could really do with a proper builder for IngestCoordinator that deals with these separate cases more neatly. I started on that in StandardIngestCoordinator but it needs more work. We don't need to do that in this PR though.

Reply:
Actually I think I'd rather avoid introducing nested ifs like this. Could this be written to use StandardIngest.builder().fromProperties(x), and then follow up by setting the fields relevant to arrow/arraylist and direct/async?